sympy/integrals/tests/test_prde.py

```diff
@@ -144,6 +146,41 @@ def test_prde_no_cancel():
     assert (Matrix([q])*V[0][:6, :])[0] == Poly(x - 1/2, t, domain='QQ(x)')
 
 
+def test_prde_cancel_liouvillian():
+    ### 1. case == 'primitive'
+    # used when integrating (log(-1/y)/2 - log(1/y)/2)/y - (log(1 - 1/y)/2 - log(1 + 1/y)/2)/y
+    # Not taken from 'the' book
+    DE = DifferentialExtension(extension={'D': [Poly(1, x), Poly(1/x, t)]})
+    assert prde_cancel_liouvillian(Poly(-1/(x - 1), t), [Poly(-x + 1, t), Poly(1, t)], 1, DE) == \
+        ([Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly((x - 1)*t, t, domain='ZZ(x)'),
+            Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'),
+            Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly(x - 1, t, domain='ZZ(x)'),
+            Poly(-x**2 + x, t, domain='ZZ(x)'), Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'), Poly(0, t, domain='QQ'),
+            Poly(0, t, domain='QQ')], Matrix([[ 0,  1,  0,  0, 0,  0,  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
```

Comment:
Maybe the list and the matrix could be stored separately. Then some tests concerning rank and/or nullspace could be done with the matrix The expression itself is not very illuminating.